src/bluesearch/entrypoint/database/parse.py

```diff
@@ -107,22 +131,45 @@ def run(
     input_type: str,
     input_path: Path,
     output_dir: Path,
+    match_filename: Optional[str],
+    recursive: bool,
+    dry_run: bool,
 ) -> int:
     """Parse one or several articles.
 
     Parameter description and potential defaults are documented inside of the
     `get_parser` function.
     """
     inputs: Iterable[Path]
+
     if input_path.is_file():
         inputs = [input_path]
+
     elif input_path.is_dir():
-        inputs = sorted(input_path.glob("*"))
+        if recursive:
+            pattern = "**/*"
+        else:
+            pattern = "*"
+        files = (x for x in input_path.glob(pattern) if x.is_file())
+
+        if match_filename:
```

Comment:
Maybe this?
```suggestion
        if match_filename is not None:
```

Reply:
The current implementation is to avoid empty patterns, i.e. `-- match-filename ""`.

Reply:
So you did it on purpose? Why does  `""` get a special treatment? It seems like 

Anyway, I thought that your goal was to simply determine whether the user provided that option or not. If they didn't then we can be sure `argparse` parses it to `None`.

Reply:
> So you did it on purpose?

Yes.

> Why does `""` get a special treatment?

To prevent users from forgetting to provide the pattern, an error is now raised in this case.
Code and test added in 77b281b440430d25b3f00be87e91588cd76fd6c9.